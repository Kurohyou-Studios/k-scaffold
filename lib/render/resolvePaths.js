const path = require('path');

const resolvePaths = (source,destination,dynamicDestination = false,entry) => {
  if(!source || !destination || !entry){
    throw('K-scaffold build error. Invalid arguments.');
  }
  const resSource = path.resolve(source,entry.name);
  const trueDestination = destination ?
    (
      dynamicDestination ?
        `${destination}/${entry.name.replace(/\.[^.]+$/,'')}` :
        destination
    ) :
    null;
  const resDest = path.resolve(trueDestination,entry.name.replace(/\.pug$/,'.html').replace(/\.scss$/,'.css'));
  return [resSource,resDest];
}

module.exports = resolvePaths;