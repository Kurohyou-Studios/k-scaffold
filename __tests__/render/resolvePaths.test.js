import { describe, it, expect, vi } from 'vitest';
import resolvePaths from './../../lib/render/resolvePaths';

describe('resolvePaths()',()=>{
  it('Should resolve the relative source and destination paths',()=>{
    const retArr = resolvePaths('./','./',false,{name:'test.pug'});
    expect(retArr[0]).toMatch(/test\.pug$/);
    expect(retArr[1]).toMatch(/test\.html$/);
  })
  it('Should resolve the relative source and destination paths dynamically.',()=>{
    const retArr = resolvePaths('./','./',true,{name:'test.pug'});
    console.log('retArr[1]:',retArr[1]);
    expect(retArr[0]).toMatch(/test\.pug$/);
    expect(retArr[1]).toMatch(/test\\test\.html$/);
  })
});