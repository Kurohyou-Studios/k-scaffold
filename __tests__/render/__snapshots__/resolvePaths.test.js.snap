// Vitest Snapshot v1

exports[`resolvePaths() > Should resolve the relative source and destination paths 1`] = `
[
  "C:\\\\Users\\\\scase\\\\OneDrive\\\\Documents\\\\git\\\\kurohyou_studios\\\\k-scaffold\\\\test.pug",
  "C:\\\\Users\\\\scase\\\\OneDrive\\\\Documents\\\\git\\\\kurohyou_studios\\\\k-scaffold\\\\test.html",
]
`;
